src/disk_io_thread.cpp

```diff
@@ -1087,10 +1104,6 @@ namespace libtorrent
 		TORRENT_ASSERT(j->next == 0);
 		TORRENT_ASSERT((j->flags & disk_io_job::in_progress) || !j->storage);
 
-		mutex::scoped_lock l(m_cache_mutex);
-
-		check_cache_level(l, completed_jobs);
-
 		DLOG("perform_job job: %s ( %s%s) piece: %d offset: %d outstanding: %d\n"
```

Comment:
Hmm, I suppose `j->d.io.offset` might be modified by another thread. On x86 at least I don't think this would do any harm, 32-bit reads are atomic anyways, but on more exotic platforms it might matter.